File: MovieMentor/Data/Movie.cs
namespace MovieMentor.Data;

public class Movie
{
    public int ID { get; set; }

    public string Title { get; set; }

    public ICollection<Genre> Genres { get; set; }

    public ICollection<Director> Directors { get; set; }

    public ICollection<Actor> Actors { get; set; }
    
    public int Year { get; set; }
    
    public ICollection<Award> Awards { get; set; }
    
    public int Duration { get; set; }
    
    public Country Country { get; set; }
    
    public double Rating { get; set; }
}

File: MovieMentor/ClientApp/src/app/home/movie.ts
import Director from "./director";
import Actor from "./actor";

export default interface Movie {
  id: number;
  title: string;
  genres: string[];
  director: Director[];
  actor: Actor[];
  year: number;
  awards: string[];
  duration: number;
  country: string;
  rating: string;
}
